src/common/types/IPlatformConfig.d.ts

```diff
@@ -27,5 +27,4 @@ export interface IPlatformConfig {
     byteLength: number,
     callback: (err: Error | null, result: ArrayBuffer | null) => void
   ) => void;
-  isWebworker?: boolean;
```

Comment:
Will do

Reply:
@lawrence-forooghian i don't see how this makes it consistent with the rest of platform config?

see:
```ts
  fetchSupported?: boolean;
  xhrSupported?: boolean;
  allowComet?: boolean;
  streamingSupported?: boolean;
```

IMO, the types are just as descriptive with the property being optional but with the additional bonus that we don't bloat other platforms with irrelevant config

Reply:
@owenpearson after taking a second look on this I do agree that it would be more consistent to leave it optional.

technically, it is a bit more error-prone when some configs are optional - you can forget to set all necessary configs when adding a new platform. but adding a new platform is almost never a thing, so I would say it's not an issue to justify bloating.

however, we can still make IPlatformConfig a bit more self-explanatory for future cases like this. so it's obvious if a config property should be optional or required.

how about doing it like this:
```javascript
export interface IPlatformCommonConfig {
  agent: string;
  logTimestamps: boolean;
  binaryType: BinaryType;
  ... and all other non-optional properties
}

export interface IPlatformSpecificConfig {
  fetchSupported?: boolean;
  xhrSupported?: boolean;
  allowComet?: boolean;
  streamingSupported?: boolean;
  ... all other optional properties
}

export type IPlatformConfig = IPlatformCommonConfig & IPlatformSpecificConfig;
```

with some jsdocs explaining the purpose of interfaces.

we can even make it more explicit if we want to:
```javascript
export type IPlatformConfig = Required<IPlatformCommonConfig> & Partial<IPlatformSpecificConfig>;
```

what do you think?

Reply:
I have no idea why I suggested making it non-optional, to be honest. You're right, it should remain optional. My "consistency" comment was just meant to be about keeping the property in the platform config instead of moving it to the `utils` module.

As for your suggestion @VeskeR, seems sensible but I think outside the scope of this PR.

Reply:
Changed `isWebworker` back to be optional.
Implemented my suggestions regarding IPlatformConfig in https://github.com/ably/ably-js/pull/1563